server/src/test/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesActionTests.java

```diff
@@ -139,7 +139,7 @@ public void handleResponse(GetDiscoveredNodesResponse response) {
             public void handleException(TransportException exp) {
                 final Throwable rootCause = exp.getRootCause();
                 assertThat(rootCause, instanceOf(IllegalArgumentException.class));
-                assertThat(rootCause.getMessage(), equalTo("discovered nodes are not exposed by discovery type [zen]"));
+                assertThat(rootCause.getMessage(), equalTo("discovered nodes are not exposed by discovery type [zen2]"));
```

Comment:
Similarly, discovered nodes _are only_ supported by discovery type `zen2` so this seems wrong. I think the right fix is to pass another discovery type into the `TransportGetDiscoveredNodesAction` constructor above, choosing randomly from `zen`, `single-node` and some other random string.